R/AutoTuner.R

```diff
@@ -99,20 +110,25 @@ AutoTuner = R6Class("AutoTuner",
     #'
     #' @param tuner ([Tuner])\cr
     #' Tuning algorithm to run.
-    initialize = function(learner, resampling, measure, search_space,
-      terminator, tuner, store_tuning_instance = TRUE,
+    initialize = function(learner, resampling, measure, terminator, tuner,
+      search_space = NULL, store_tuning_instance = TRUE,
       store_benchmark_result = TRUE, store_models = FALSE,
       check_values = FALSE) {
+      learner = assert_learner(learner)$clone(deep = TRUE)
+
+      if (!is.null(search_space) && length(learner$param_set$get_values(type = "only_token")) > 0) {
+        stop("If the values of the ParamSet of the Learner contain TuneTokens you cannot supply a search_space.")
+      }
+      if (is.null(search_space)) {
+        search_space = learner$param_set$search_space()
+        learner$param_set$values = learner$param_set$get_values(type = "without_token")
```

Comment:
you cannot do this here because this unsets parameters that may be required. Instead you should be doing this inside the tuning code.

One option here would be to leave search_space `NULL`, to leave the `learner$param_set` as it is, and trust the `TuningInstance` initialization code to take care of this.

Reply:
Another way would be to have a "placeholder" token, similar to TuneToken, that prevents the `"requires"` check from triggering but has no other functionality. `TuneToken` could even inherit from `PlaceholderToken`. The semantics would be that whenever a `ParamSet$values` slot contains a `PlaceholderToken`, then its `"requires"` are fulfilled, its dependencies are fulfilled, but calling `Learner$train()` should throw an error (because the placeholder would be replaced by a concrete value in the tuning code). I.e. `PlaceholderToken` would be doing some part of what `TuneToken` is doing already, with the exception that `PlaceholderToken` does not generate a `search_space`.